src/components/IstioWizards/IstioWizard.tsx

```diff
@@ -99,15 +85,44 @@ class IstioWizard extends React.Component<Props, State> {
     return true;
   };
 
-  componentDidMount() {
-    this.resetState();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    if (prevProps.show !== this.props.show || !this.compareWorkloads(prevProps.workloads, this.props.workloads)) {
-      this.resetState();
+  buildHTTPMatchRequest = (matches: string[]): HTTPMatchRequest[] => {
+    const matchRequests: HTTPMatchRequest[] = [];
+    const matchHeaders: HTTPMatchRequest = { headers: {} };
+    // Headers are grouped
+    matches
+      .filter(match => match.startsWith('headers'))
+      .forEach(match => {
+        // match follows format:  headers [<header-name>] <op> <value>
+        const i0 = match.indexOf('[');
+        const j0 = match.indexOf(']');
+        const headerName = match.substring(i0 + 1, j0);
+        const i1 = match.indexOf(' ', j0 + 1);
+        const j1 = match.indexOf(' ', i1 + 1);
+        const op = match.substring(i1 + 1, j1);
+        const value = match.substring(j1 + 1);
+        matchHeaders.headers![headerName] = { [op]: value };
+      });
+    if (Object.keys(matchHeaders.headers || {}).length > 0) {
+      matchRequests.push(matchHeaders);
     }
-  }
+    // Rest of matches
+    matches
+      .filter(match => !match.startsWith('headers'))
+      .forEach(match => {
+        // match follows format: <name> <op> <value>
+        const i = match.indexOf(' ');
+        const j = match.indexOf(' ', i + 1);
+        const name = match.substring(0, i);
+        const op = match.substring(i, j);
+        const value = match.substring(j);
+        matchRequests.push({
```

Comment:
The same here than above.

Reply:
Yeap. Good suggestion, but as commented, <value> can be also a regex and then it may lead in more complexity.